lib/Dialect/FIRRTL/Transforms/LowerTypes.cpp

```diff
@@ -1080,21 +1080,29 @@ bool TypeLoweringVisitor::visitDecl(MemOp op) {
             mkConnect(builder, realOldField, newField);
           } else {
             // Cast the input aggregate write data to flat type.
-            realOldField = builder->create<BitCastOp>(
-                newField.getType().cast<FIRRTLType>(), oldField);
+            auto newFieldType = newField.getType().cast<FIRRTLType>();
+            auto oldFieldBitWidth = getBitWidth(oldField.getType());
+            // Following condition is true, if a data field is 0 bits. Then
+            // newFieldType is of smaller bits than old.
+            if (getBitWidth(newFieldType) != oldFieldBitWidth.getValue())
+              newFieldType =
+                  UIntType::get(context, oldFieldBitWidth.getValue());
```

Comment:
Checking it this way as opposed to a more explicit walk seems a little janky. A utility looking for zero-width fields in an aggregate seems cleaner. I'm alright with the implementation in the PR given the comment.